gslib/metrics.py

```diff
@@ -728,19 +728,25 @@ def ReportMetrics(
     if log_file_path is not None:
       # If the path is not None, we'll need to surround the path with quotes
       # so that the path is passed as a string to the metrics_reporter module.
-      log_file_path = '"%s"' % log_file_path
-
-    reporting_code = ('from gslib.metrics_reporter import ReportMetrics; '
-                      'ReportMetrics("{0}", {1}, log_file_path={2})').format(
-                          temp_metrics_file.name,
-                          log_level,
-                          log_file_path).encode('utf-8').decode('unicode_escape')
+      log_file_path = 'r"%s"' % log_file_path
+
+    reporting_code = six.ensure_str(
```

Comment:
Q: Why is the `ensure_str` call needed here?  Are we worried about any of the types we're plugging into the string being the `bytes` type?

Having `unicode_literals` imported means that the format string should be a unicode type, right?  In general, to avoid errors that come from type-mismatches (like adding a bytes type into a unicode format string and getting something like `my_value=b'abc'` instead of `my_value=abc`), I prefer to convert each of the arguments going into the format string to the `str` type first, rather than feeding them all to the format string and then casting the result to a string.